api/src/main/java/org/pmiops/workbench/db/model/DbInstitution.java

```diff
@@ -78,21 +94,69 @@ public void setOrganizationTypeOtherText(String organizationTypeOtherText) {
   public Set<DbInstitutionEmailDomain> getEmailDomains() {
     return emailDomains;
   }
 
-  public void setEmailDomains(Iterable<DbInstitutionEmailDomain> emailDomains) {
-    this.emailDomains =
-        StreamSupport.stream(emailDomains.spliterator(), false).collect(Collectors.toSet());
+  /**
+   * Hibernate doesn't like it when you reassign collections. Instead, modify in-place. First, call
+   * retainAll() to subset DB rows to those we wish to keep: the intersection of old and new. Then,
+   * call addAll() to add the diff(new - old) rows.
+   *
+   * <p>https://stackoverflow.com/questions/5587482/hibernate-a-collection-with-cascade-all-delete-orphan-was-no-longer-referenc
+   *
+   * @param emailDomains the new set of domains for this Institution
+   */
+  public void setEmailDomains(final Set<DbInstitutionEmailDomain> emailDomains) {
+    final Set<DbInstitutionEmailDomain> attachedDomains =
+        emailDomains.stream()
+            .map(address -> address.setInstitution(this))
+            .collect(Collectors.toSet());
+    // modifies this set so that its value is the intersection of the two sets
+    this.emailDomains.retainAll(attachedDomains);
+    this.emailDomains.addAll(Sets.difference(attachedDomains, this.emailDomains));
   }
 
+  /** Builder method to help streamline the building of a DbInstitution. */
+  public DbInstitution emailDomains(Set<DbInstitutionEmailDomain> emailDomains) {
+    setEmailDomains(emailDomains);
+    return this;
+  }
+
+  @OneToMany(mappedBy = "institution", cascade = CascadeType.ALL, fetch = FetchType.EAGER)
   public Set<DbInstitutionEmailAddress> getEmailAddresses() {
     return emailAddresses;
   }
 
-  public void setEmailAddresses(Iterable<DbInstitutionEmailAddress> emailAddresses) {
-    this.emailAddresses =
-        StreamSupport.stream(emailAddresses.spliterator(), false).collect(Collectors.toSet());
+  /**
+   * Hibernate doesn't like it when you reassign collections. Instead, modify in-place. First, call
+   * retainAll() to subset DB rows to those we wish to keep: the intersection of old and new. Then,
+   * call addAll() to add the diff(new - old) rows.
+   *
+   * <p>https://stackoverflow.com/questions/5587482/hibernate-a-collection-with-cascade-all-delete-orphan-was-no-longer-referenc
+   *
+   * @param emailAddresses the new set of addresses for this Institution
+   */
+  public void setEmailAddresses(final Set<DbInstitutionEmailAddress> emailAddresses) {
+    final Set<DbInstitutionEmailAddress> attachedAddresses =
+        emailAddresses.stream()
+            .map(address -> address.setInstitution(this))
+            .collect(Collectors.toSet());
+    // modifies this set so that its value is the intersection of the two sets
+    this.emailAddresses.retainAll(attachedAddresses);
+    this.emailAddresses.addAll(Sets.difference(attachedAddresses, this.emailAddresses));
+  }
+
+  /** Builder method to help streamline the building of a DbInstitution. */
```

Comment:
For clarity, I would make a public nested Builder class and put this there? An example is [here](https://github.com/all-of-us/workbench/blob/master/api/src/main/java/org/pmiops/workbench/monitoring/MeasurementBundle.java#L70). Though I guess you still need your no-arg ctor 🤔 .

Anyway, I wouldn't make an alias function (a non-overload with the same body) unless there's a compelling reason.

Reply:
Merged these to avoid confusion